src/store/flow/block-types/MobilePrimitives_NumericResponseBlockStore.ts

```diff
@@ -72,70 +109,58 @@ const actions: ActionTree<IEmptyState, IRootState> = {
-    const validationMin = (block.config as INumericBlockConfig)?.validation_minimum
-
-    // validationMin & validationMax relations, valid for all modes
-    if ((validationMax != null && validationMin != null && validationMax < validationMin)
-      || (validationMax == null && validationMin != null)) {
-      validationResults.ajvErrors.push({
-        dataPath: '/config/validation_minimum',
-        message: lang.trans('flow-builder-validation.numeric-block-min-value-must-be-lower-than-max-value'),
-      } as ErrorObject)
-      validationResults.ajvErrors.push({
-        dataPath: '/config/validation_maximum',
-        message: lang.trans('flow-builder-validation.numeric-block-max-value-must-be-greater-than-min-value'),
-      } as ErrorObject)
+    // validationMax must be greater than validationMin
+    if (isMinGreaterThanMax || hasMinOnly) {
+      errors
+        .add(CONFIG_VALIDATION_MINIMUM, 'numeric-block-min-value-must-be-lower-than-max-value')
+        .add(CONFIG_VALIDATION_MAXIMUM, 'numeric-block-max-value-must-be-greater-than-min-value')
     }
 
-    // MaxDigit & validationMin/validationMax relations, valid for iVR only
+    // IVR only checks
     if (rootGetters['flow/hasVoiceMode'] as boolean) {
-      // Must have one of MaxDigit or validationMax
-      if (maxDigits == null && validationMax == null) {
-        validationResults.ajvErrors.push({
-          dataPath: '/config/ivr/max_digits',
-          message: lang.trans('flow-builder-validation.numeric-block-missing-max-value-and-max-digits'),
-        } as ErrorObject)
-        validationResults.ajvErrors.push({
-          dataPath: '/config/validation_maximum',
-          message: lang.trans('flow-builder-validation.numeric-block-missing-max-value-and-max-digits'),
-        } as ErrorObject)
+      if (!hasMax && !hasMaxDigits) {
+        // Either maxDigits or validationMax must be defined
+        errors
+          .add(CONFIG_MAX_DIGITS, 'numeric-block-missing-max-value-and-max-digits')
+          .add(CONFIG_VALIDATION_MAXIMUM, 'numeric-block-missing-max-value-and-max-digits')
       }
 
-      if (maxDigits != null) {
-        // validationMin must comply with MaxDigit
-        if (validationMin != null && maxDigits * 9 < validationMin) {
-          validationResults.ajvErrors.push({
-            dataPath: '/config/ivr/max_digits',
-            message: lang.trans('flow-builder-validation.numeric-block-min-value-must-be-lower-than-implied-max-digits'),
-          } as ErrorObject)
-          validationResults.ajvErrors.push({
-            dataPath: '/config/validation_minimum',
-            message: lang.trans('flow-builder-validation.numeric-block-min-value-must-be-lower-than-implied-max-digits'),
-          } as ErrorObject)
-        }
+      // Both validationMin and validationMax must have no more digits than allowed
+      if (hasMaxDigits) {
+        const maxValue = 10 ** maxDigits - 1
```

Comment:
I noticed having `maxDigits = 2` the validation won't accept values over `18` which doesn't feel right.

cc: @pascaleli

Reply:
@object-required , great catch here. Thank you very much 🙏

Reply:
@object-required , great catch here. Thank you very much 🙏